src/Directory.Build.targets

```diff
@@ -2,6 +2,29 @@
 <Project>
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
   
+  <!--
+    Set InformationalVersion to only include the original version prefix and any prerelease suffix.
+    It will be updated later with commit hash information. Without this setting to the original
+    version prefix, the version would include a patch number that is computed from the build number,
+    which makes it difficult to trace back to the exact release version. This information is also
+    used when invoking the global version option for tools using System.CommandLine.
+  -->
+  <PropertyGroup>
+    <InformationalVersion>$(_OriginalVersionPrefix)</InformationalVersion>
+    <InformationalVersion Condition="'$(VersionSuffix)' != ''">$(InformationalVersion)-$(VersionSuffix)</InformationalVersion>
+  </PropertyGroup>
+
+  <!--
+    Set PackageVersion to only include the original version prefix and any prerelease suffix.
+    Without this setting to the original version prefix, the version would include a patch number
+    that is computed from the build number, which makes it difficult to trace back to the exact
+    release version. Only set this if PackageVersion was not explicitly set.
+  -->
+  <PropertyGroup Condition="'$(PackageVersion)' == ''">
+    <PackageVersion>$(_OriginalVersionPrefix)</PackageVersion>
```

Comment:
Can we use a non-private property here? if this doesn't change from what we set originally, we could just generate a copy property for us to use.

Also, `PreReleaseVersionLabel` set to blank implies you want to generate that package version scheme. Usually we don't remove PreReleaseVersionLabel for box releases (set it to rtm or servicing). 

Instead, if you set `IsShipping` for monitor, and set DotNetFinalVersionKind the package would look like `5.0.0`.
So:
- Previews (like the upcoming ones), do nothing and you end up with: 
  - AssemblyFile:  5.0.21.18101
  - PackageVersion: dotnet-monitor.5.0.0-preview.4.21181.1.nupkg
  - AssemblyInformational: 5.0.0-preview.4.21181.1+b223bea10a931df4f993727a593321655d95dd30
  - Assembly: 5.0.0.0
- Releases by setting DotNetFinalVersionKind=release and PreReleaseVersionLabel=non-blank : 
  - AssemblyFile:  5.0.21.18101
  - PackageVersion: dotnet-monitor.5.0.0.nupkg
  - AssemblyInformational: 5.0.0+b223bea10a931df4f993727a593321655d95dd30
  - Assembly: 5.0.0.0

And the `--version` flag just returns whatever the informational version is. Is that enough? That way we don't need any custom logic.

Reply:
Thanks for that information! Seems to be working as expected, so I'll close this PR. When I was attempting to figure this out previously, it was counter-intuitive to specify `IsShipping=true`, `DotNetFinalVersionKind=true`, and `PreReleaseVersionLabel=<non-empty>`.

Reply:
Yeah, prerelease to non-empty is the crazy part... And for release you set `DotNetFinalVersionKind` to `release`. For previews it can remain empty.